.github/workflows/test.yml

```diff
@@ -11,12 +11,14 @@ on:
 
 jobs:
   test:
-    runs-on: ${{ matrix.os }}
+    runs-on: ${{ matrix.os || 'ubuntu-latest' }}
     strategy:
       fail-fast: false
       matrix:
-        python: ["3.7", "3.8", "3.9", "3.10", "pypy-3.8"]
-        os: [ubuntu-latest, windows-latest]
+        python: ["3.7", "3.8", "3.9", "3.10","~3.11.0-0", "pypy-3.8"]
+        os:
+          - ubuntu-latest
+          - windows-latest
```

Comment:
I think reverting this change is better than having to add:

```
runs-on: ${{ matrix.os || 'ubuntu-latest' }}
```

Reply:
That is the second place where I raised my brows while testing the pipeline, GHA complained about os at this line, and making it an explicit list fixed the issue. It is bit perplexing because I know well enough that part of the YAML specification and both representations are valid and should load the same. Still, one failed another one did not.

Probably you know the Github is not using a standard YAML loading library and they have their own implementation, one that does not even supports anchors. I would not be surprised to be another issue related to that.

PS. `||` works even for undefined values, if the value on the left side is not defined, it will pick the second value.

You are free to try other variants including modifying this pr.